biodata-formats/src/test/java/org/opencb/biodata/formats/variant/vcf4/io/VariantVcfReaderTest.java

```diff
@@ -43,6 +54,101 @@ public void readFile() {
     }
 
     @Test
+    public void readFileVersion4_1() {
+        String inputFile = getClass().getResource("/variant-test-file-4.1.vcf.gz").getFile();
+        VariantSource source = new VariantSource(inputFile, "test", "test", "Test file");
+
+        VariantReader reader = new VariantVcfReader(source, inputFile);
+
+        List<Variant> variants;
+
+        assertTrue(reader.open());
+        assertTrue(reader.pre());
+
+        int i = 0;
+        do {
+            variants = reader.read();
+            if(variants != null){
+                i+=variants.size();
+            }
+        } while (variants != null);
+
+        assertEquals(753, i);
+
+        assertTrue(reader.post());
+        assertTrue(reader.close());
+    }
+
+    @Test
+    public void testHeaderVersion4_1() {
+        String inputFile = getClass().getResource("/variant-test-file-4.1.vcf.gz").getFile();
+        VariantSource source = new VariantSource(inputFile, "test", "test", "Test file");
+
+        VariantReader reader = new VariantVcfReader(source, inputFile);
+
+        assertTrue(reader.open());
+        assertTrue(reader.pre());
+        assertTrue(reader.post());
+        assertTrue(reader.close());
+
+        assertNotNull(source.getMetadata().get("contig"));
+        assertFalse(((Collection)source.getMetadata().get("contig")).isEmpty());
+        assertTrue(((Collection)source.getMetadata().get("contig")).iterator().next() instanceof String);
+        assertNotNull(source.getMetadata().get("FILTER"));
+        assertFalse(((Collection)source.getMetadata().get("FILTER")).isEmpty());
+        assertTrue(((Collection)source.getMetadata().get("FILTER")).iterator().next() instanceof VcfFilterHeader);
+        assertNotNull(source.getMetadata().get("ALT"));
+        assertFalse(((Collection)source.getMetadata().get("ALT")).isEmpty());
+        assertTrue(((Collection)source.getMetadata().get("ALT")).iterator().next() instanceof VcfAlternateHeader);
+        assertNotNull(source.getMetadata().get("FORMAT"));
+        assertFalse(((Collection)source.getMetadata().get("FORMAT")).isEmpty());
+        assertTrue(((Collection)source.getMetadata().get("FORMAT")).iterator().next() instanceof VcfFormatHeader);
+        assertNotNull(source.getMetadata().get("INFO"));
+        assertFalse(((Collection)source.getMetadata().get("INFO")).isEmpty());
+        assertTrue(((Collection)source.getMetadata().get("INFO")).iterator().next() instanceof VcfInfoHeader);
```

Comment:
Are there multiple instances of any of contig, FILTER, ALT, etc? If so it would be good to have a loop to know that the correct number of them are processed.